services/vision/server.go

```diff
@@ -217,6 +200,54 @@ func segmentsToProto(frame string, segs []*vision.Object) ([]*commonpb.PointClou
 	return protoSegs, nil
 }
 
+func (server *serviceServer) CaptureAllFromCamera(
+	ctx context.Context,
+	req *pb.CaptureAllFromCameraRequest,
+) (*pb.CaptureAllFromCameraResponse, error) {
+	ctx, span := trace.StartSpan(ctx, "service::vision::server::CaptureAllFromCamera")
+	defer span.End()
+	svc, err := server.coll.Resource(req.Name)
+	if err != nil {
+		return nil, err
+	}
+	capt, err := svc.CaptureAllFromCamera(ctx,
+		req.CameraName,
+		req.ReturnImage,
+		req.ReturnDetections,
+		req.ReturnClassifications,
+		req.ReturnObjectPointClouds,
+		req.Extra.AsMap(),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	objProto, err := segmentsToProto(req.CameraName, capt.PointCloudObject())
+	if err != nil {
+		return nil, err
+	}
+
+	imgProto, err := imageToProto(ctx, capt.Image(), utils.MimeTypeJPEG)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.CaptureAllFromCameraResponse{
+		Image:           imgProto,
+		Detections:      detsToProto(capt.Detections()),
+		Classifications: clasToProto(capt.Classifications()),
+		Objects:         objProto,
+	}, nil
+}
+
+func imageToProto(ctx context.Context, img image.Image, mimeType string) (*v11.Image, error) {
+	imgBytes, err := rimage.EncodeImage(ctx, img, mimeType)
+	if err != nil {
+		return nil, err
+	}
+	return &v11.Image{Image: imgBytes}, nil
```

Comment:
Also, Image has more fields than "Image", you should also store the"Format" that you ended up encoding it as. You also have the original camera name -- put that in the "SourceName" field